pf/internal/plugin/provider_server.go

```diff
@@ -235,7 +229,18 @@ func (p *providerServer) CheckConfig(ctx context.Context,
 		return nil, p.checkNYI("CheckConfig", err)
 	}
 
-	rpcInputs, err := p.configEncoding.MarshalProperties(newInputs)
+	// The bridge (and [providerServer] more specifically) attempts to respect the
+	// hand-shake in performed in [Configure]. Unfortunately, [CheckConfig] happens
+	// before [Configure] so we just assume that the engine supports secrets.
```

Comment:
I think this is reasonable. This entire file I understand will go away in favor of linking code directly from p/p at some point.